### ISSUE TITLE
Proposal to reduce confusing behaviors and improve consistency with other plugins

### ISSUE THREAD

**Issue by jordansissel** [logstash-plugins/logstash-input-jdbc#207]
Some settings changes:

#### rename `statement_filepath`

Problem: "filepath" feels redundant. 

Proposal: Rename to `statement_path`

#### remove `use_column_value` 

Problem: You can specify `tracking_column` with intent to use a column as your `sql_last_value`. However, this does nothing alone and is very confusing to me. To make this actually work, you have to also specify `use_column_value => true`.

Proposal: Infer this from if `tracking_column` is set at all. If the user specifies any value for `tracking_column` this should express enough intent to use that column as the value. If `@tracking_column` is non-nil, we should use that column's value for `sql_last_value`

#### remove `tracking_column_type`

Problem: If you specify the wrong one, you can get an error.

Proposal: We can infer the column type from the database, can't we?

#### remove `clean_run`

Problem: I don't understand the use case. This setting indicates a kind of management task, and we try not to do management tasks in plugins. 

Proposal: Removing the file outside of logstash seems the correct approach.

#### rename `jdbc_user`

Problem: jdbc_ prefix on things is perhaps unnecessary. We know we are in the jdbc plugin already.

Proposal: rename to `user`

#### rename `jdbc_password`

Problem: see above

Proposal: rename to `password`

#### remove `jdbc_password_filepath`

Problem: this is inconsistent with any other password setting we have.

Proposal: Remove it. We'll be able to specify password from files again once we have https://github.com/elastic/logstash/issues/6892 implemented.

#### reconsider `jdbc_validate_connection`

Problem: I don't understand why a user would want to turn this setting off.

Proposal: remove this setting.

#### remove `jdbc_validation_timeout`

Problem: See above.

Proposal: remove this setting.

#### remove `connection_retry_attempts`

Problem: Why would we want to give up connecting to the database?

Proposal: remove this  setting.

#### remove `connection_retry_attempts_wait_time`

Problem: users should not concern themselves with such details.

Proposal: remove this setting. Replace behavior with exponential backoff.

**Comment by andrewvc** [comment]
I like most of this stuff. I think the `fetch_size` and `paging_enabled` options might be about whether you want to use a cursor vs. multiple statements.

I could be wrong, but if that is the case we might want a `strategy` option that can be set to `cursor` or `multiple_statements`.

**Comment by jordansissel** [comment]
@andrewvc I agree that if it turns out we do want to provide cursor-vs-multiple-statements, that exposing that as a strategy selection is an improvement over the "fetch_size" vs "page_size+paging_enabled" settings.

**Comment by acchen97** [comment]
@jordansissel thanks for the review here. Although I agree with most of the suggestions listed here, this isn't the highest priority at this time. I'd suggest we revisit this sometime following the next major.
